modules/fedimint-mint-client/src/issuance.rs

```diff
@@ -0,0 +1,372 @@
+use std::time::Duration;
+
+use fedimint_client::sm::{ClientSMDatabaseTransaction, OperationId, State, StateTransition};
+use fedimint_client::DynGlobalClientContext;
+use fedimint_core::api::GlobalFederationApi;
+use fedimint_core::encoding::{Decodable, Encodable};
+use fedimint_core::module::registry::ModuleDecoderRegistry;
+use fedimint_core::{Amount, OutPoint, Tiered, TieredMulti, TransactionId};
+use fedimint_derive_secret::{ChildId, DerivableSecret};
+use fedimint_mint_common::{BlindNonce, MintOutputBlindSignatures, MintOutputOutcome, Nonce, Note};
+use secp256k1::{KeyPair, Secp256k1, Signing};
+use serde::{Deserialize, Serialize};
+use tbs::{blind_message, unblind_signature, AggregatePublicKey, BlindedSignature, BlindingKey};
+use thiserror::Error;
+use tracing::error;
+
+use crate::db::NoteKey;
+use crate::{MintClientContext, SpendableNote};
+
+/// State machine managing the e-cash issuance process related to a mint output.
+///
+/// ```mermaid
+/// graph LR
+///     classDef virtual fill:#fff,stroke-dasharray: 5 5
+///
+///     Created -- containing tx rejected --> Aborted
+///     Created -- await output outcome --> Outcome["Outcome Received"]:::virtual
+///     subgraph Await Outcome
+///     Outcome -- valid blind signatures  --> Succeeded
+///     Outcome -- invalid blind signatures  --> Failed
```

Comment:
What if the dishonest federation never provides any blind signatures?

Reply:
If the federation is operational this will block forever, if it's offline I assume we enter a failure state.